### PR TITLE
Add onchain models

### PR DESCRIPTION
**Pull request by motechFR**

### CODE REVIEW

File: src/prisma/schema.prisma [hunk 1]
```diff
@@ -2839,6 +2839,7 @@ model Scout {
   mergedFromEvents                 ScoutMergeEvent[]            @relation("mergedFromScout")
   mergedToEvents                   ScoutMergeEvent[]            @relation("mergedToScout")
   PartnerRewardEvent               PartnerRewardEvent[]
+  scoutedNFTs                         ScoutNFT[]
```
Comment (mattcasey) [inline src/prisma/schema.prisma:2842]: We've been using "Nft" in most places, i think we should stay consistent

File: src/prisma/schema.prisma [hunk 2]
```diff
@@ -2881,6 +2882,7 @@ enum BuilderEventType {
   nft_purchase
   proposal_passed
   gems_payout
+  onchain_gems_payout
```
Comment (mattcasey) [inline src/prisma/schema.prisma:2885]: why do we need a new gems_payout?

File: src/prisma/schema.prisma [hunk 3]
```diff
@@ -3261,23 +3264,40 @@ model WeeklyClaims {
...
-  senderId    String?             @db.Uuid
-  sender      Scout?              @relation(fields: [senderId], references: [id], onDelete: Cascade, name: "tokenReceiptSender")
-  activities  ScoutGameActivity[]
+  id            String              @id @default(uuid()) @db.Uuid
+  createdAt     DateTime            @default(now())
+  value         Int
+  claimedAt     DateTime?
+  eventId       String              @db.Uuid
+  event         BuilderEvent        @relation(fields: [eventId], references: [id], onDelete: Cascade)
+  walletAddress String
```
Comment (mattcasey) [inline src/prisma/schema.prisma:3273]: this should have a foreign key relationship to our ScoutWallet collection

File: src/prisma/schema.prisma [hunk 4]
```diff
@@ -3261,23 +3264,40 @@ model WeeklyClaims {
...
   @@index([eventId])
 }
 
+model ScoutNFT {
+  id           String     @id @default(uuid()) @db.Uuid
+  createdAt    DateTime   @default(now())
+  updatedAt    DateTime   @updatedAt
+  builderNftId String     @db.Uuid
+  builderNft   BuilderNft @relation(fields: [builderNftId], references: [id], onDelete: Cascade)
+  scoutId      String     @db.Uuid
```
Comment (mattcasey) [inline src/prisma/schema.prisma:3291]: this needs to connect to a ScoutWallet and not a scout directly

File: src/prisma/schema.prisma [hunk 5]
```diff
@@ -3261,23 +3264,40 @@ model WeeklyClaims {
...
+  createdAt     DateTime            @default(now())
+  value         Int
+  claimedAt     DateTime?
+  eventId       String              @db.Uuid
+  event         BuilderEvent        @relation(fields: [eventId], references: [id], onDelete: Cascade)
+  walletAddress String
+  recipientId   String?             @db.Uuid
+  recipient     Scout?              @relation(fields: [recipientId], references: [id], onDelete: Cascade, name: "tokenReceiptRecipient")
+  senderId      String?             @db.Uuid
+  sender        Scout?              @relation(fields: [senderId], references: [id], onDelete: Cascade, name: "tokenReceiptSender")
```
Comment (mattcasey) [inline src/prisma/schema.prisma:3277]: replace receipitnId/senderId with recipientWallet and senderWallet and connect them to ScoutWallet

**Review by mattcasey — CHANGES_REQUESTED** [review]
see comments

File: src/prisma/schema.prisma [hunk 6]
```diff
@@ -3261,23 +3264,40 @@ model WeeklyClaims {
...
-  event       BuilderEvent        @relation(fields: [eventId], references: [id], onDelete: Cascade)
-  recipientId String?             @db.Uuid
-  recipient   Scout?              @relation(fields: [recipientId], references: [id], onDelete: Cascade, name: "tokenReceiptRecipient")
-  senderId    String?             @db.Uuid
-  sender      Scout?              @relation(fields: [senderId], references: [id], onDelete: Cascade, name: "tokenReceiptSender")
-  activities  ScoutGameActivity[]
+  id            String              @id @default(uuid()) @db.Uuid
+  createdAt     DateTime            @default(now())
+  value         Int
+  claimedAt     DateTime?
```
Comment (mattcasey) [inline src/prisma/schema.prisma:3270]: might be good to add a transactionhash to show where it was claimed?

**Comment by motechFR** [comment]
Thanks @mattcasey I've addressed all the feedback raised

**Review by mattcasey — APPROVED** [review]

**Pull request merged**